lib/ansible/compat/__init__.py

```diff
@@ -1,26 +0,0 @@
-# (c) 2014, Toshio Kuratomi <tkuratomi@ansible.com>
-#
-# This file is part of Ansible
-#
-# Ansible is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# Ansible is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
-
-# Make coding more python3-ish
-from __future__ import (absolute_import, division, print_function)
-__metaclass__ = type
-
-'''
-Compat library for ansible.  This contains compatibility definitions for older python
-When we need to import a module differently depending on python version, do it
-here.  Then in the code we can simply import from compat in order to get what we want.
-'''
```

Comment:
This file should probably not have been deleted/truncated.

Reply:
Although, I suppose that we could make it a policy that any backwards compatibility code should go into module_utils so it is usable module-side as well as controller side.

@sivel, @sdoran ^ Do you think it's okay to get rid of the controller side compat and make it policy that all compat code goes into module_utils from now on?

Reply:
Ah... I see @mattclay commented that there should be a backwards-compat shim so third party plugins (especially third party connection plugins that may have based their work on the connection plugins we ship) will keep working.  That seems like a good idea and would mean that `compat/__init__.py` needs to remain for now as well.

Reply:
I think generally compat code should go into module_utils rather than being controller-only, as long as it's likely to be usable by modules.

In this case, since we're moving existing compat code we should be sure to keep a shim in the original location for backwards compatibility with existing plugins. We may want a deprecation warning for use of that shim though, so we don't have to maintain it indefinitely.

Reply:
PS: The backwards compat shim will look something like the code in ib/ansible/module_utils/compat/selectors/__init__.py